llmfoundry/command_utils/data_prep/split_eval_set.py

```diff
@@ -0,0 +1,185 @@
+# Copyright 2024 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import os
+import re
+import json
+import contextlib
+import datasets as hf_datasets
+import numpy as np
+from typing import Optional
+
+import composer.utils as utils
+from llmfoundry.data.finetuning.tasks import maybe_safe_download_hf_data
+
+log = logging.getLogger(__name__)
+
+DELTA_JSONL_REGEX = re.compile(r"^tmp-t$")
```

Comment:
probably better to not hardcode this and just have your functionality accept an arbitrary local path?

Reply:
Some notes:
1. These regexes are copied from MAPI and its logic
2. The issue is that the HF regex is quite generic so if we don't specify a Delta case here (by assuming a specific prefix) and check it before HF, it will mistakenly match on the HF case. This problem only occurs at this step because we convert Delta table to a local JSONL file and thus mess up its pathname formatting. The logic for processing HF datasets and local JSONL files differs because HF datasets have us look in path `foo/bar/split/file.parquet` or something and local JSONL file assumes `foo/bar/split.jsonl`.

Reply:
i changed this to be a string match on `"tmp-t"` instead of a regex, seemed simpler